storage/README.md

```diff
@@ -19,45 +19,57 @@ google-cloud-default = { version = <version>, features = ["storage"] }
 
 ```rust
 use google_cloud_storage::client::Client;
-use google_cloud_storage::sign::SignedURLOptions;
-use google_cloud_storage::sign::SignedURLMethod;
+use google_cloud_storage::client::ClientConfig;
+use google_cloud_storage::sign::{SignedURLConfig, SignedURLConfigOptions, SignedURLMethod, SignBy};
 use google_cloud_storage::http::Error;
 use google_cloud_storage::http::objects::download::Range;
 use google_cloud_storage::http::objects::get::GetObjectRequest;
-use google_cloud_storage::http::objects::upload::UploadObjectRequest;
+use google_cloud_storage::http::objects::upload::{Media, UploadObjectRequest, UploadType};
 use tokio::task::JoinHandle;
 use std::fs::File;
 use std::io::BufReader;
 use std::io::Read;
-use google_cloud_default::WithAuthExt;
-
-#[tokio::main]
-async fn main() -> Result<(), Error> {
 
+// use google_cloud_default::WithAuthExt;
+// let config = ClientConfig::default().with_auth().await?;
+async fn run(config: ClientConfig) -> Result<(), Error> {
     // Create client.
-    let config = ClientConfig::default().with_auth().await.unwrap();
     let mut client = Client::new(config);
 
     // Upload the file
     let upload_type = UploadType::Simple(Media::new("file.png"));
     let uploaded = client.upload_object(&UploadObjectRequest {
         bucket: "bucket".to_string(),
         ..Default::default()
-    }, "hello world".as_bytes(), upload_type).await;
+    }, "hello world".as_bytes(), &upload_type).await;
 
     // Download the file
     let data = client.download_object(&GetObjectRequest {
         bucket: "bucket".to_string(),
         object: "file.png".to_string(),
         ..Default::default()
-   }, &Range::default()).await;
-
-    // Create signed url.
-    let url_for_download = client.signed_url("bucket", "foo.txt", SignedURLOptions::default());
-    let url_for_upload = client.signed_url("bucket", "foo.txt", SignedURLOptions {
+    }, &Range::default()).await;
+    
+    // Create signed download url
+    let download_url_config = SignedURLConfig {
+        method: SignedURLMethod::GET,
+        google_access_id: "google_access_id".to_string(),
+        // add your private key here
+        sign_by: SignBy::PrivateKey(vec![]),
+        options: SignedURLConfigOptions::default()
+    };
+    let url_for_download = client.signed_url("bucket", "foo.txt", download_url_config).await;
+    
+    // Create signed upload url
+    let upload_url_config = SignedURLConfig {
         method: SignedURLMethod::PUT,
-        ..Default::default()
-    });
+        google_access_id: "google_access_id".to_string(),
+        // add your private key here
+        sign_by: SignBy::PrivateKey(vec![]),
+        options: SignedURLConfigOptions::default()
+    };
```

Comment:
You are right in pointing out that the default method of `SignedURLOptions `is half-baked.
The implementation you suggested makes it a bit difficult to create `SignedURLConfig`, so how about adding a method to generate `SignedURLConfig`  in the `Client` as follows?

```rust
impl SignedURLConfig {
   pub fn new(google_access_id: String, sign_by: SignBy) -> Self {
        Self {
            google_access_id: google_access_id,
            sign_by: sign_by,
            method: SignedURLMethod::GET,
            expires: std::time::Duration::from_secs(600),
            content_type: None,
            headers: vec![],
            query_parameters: Default::default(),
            md5: None,
            style: Box::new(PathStyle {}),
            insecure: false,
        }
  }
}  
impl Client {
    pub fn new_signed_url_config(&self) -> Result<SignedURLConfig, SignedURLError> {
        let google_access_id = self.default_google_access_id.clone()
            .ok_or(SignedURLError::InvalidOption("No default google_access_id is found"))?;
        let sign_by = self.default_sign_by.clone()
            .ok_or(SignedURLError::InvalidOption("No default sign_by was found"))?;
            Ok(Self::new(google_access_id, sign_by))
        }
        Ok(SignedURLOptions::new(google_access_id, sign_by))
    }
```

Reply:
Hmm i dont get the first example.
In my implementation the way you create a SignedUrlConfig is the following:

```rust
let opts = SignedURLConfig {
    sign_by: sign_by,
    google_access_id: google_access_id,
    method: SignedURLMethod::GET,
    options: SignedURLConfigOptions::default()
};
```

Imo using a function to create that does not really make it easier

Reply:
I tried implementing your second code snippet

Reply:
The code is more descriptive than the original and we would like to improve it.
In most cases, we want to use the `default_sign_by` and `default_google_access_id` that the `Client` holds, so we want to be able to omit them.
If you don't want to include `google_access_id` and `sign_by` in `default`, how about separating them from `SignedURLConfig`, for example?

```rust
let option = SignedURLConfig {
      method: SignedURLMethod::PUT,
      options: SignedURLConfigOptions::default()
};
client.signed_url(bucket,object, config, None, None);
//client.signed_url(bucket,object, config, Some(your_original_google_access_id), Some(your_original_sign_by))
```

Reply:
I implemented your requested changes as they make more sense then what i was doing.
I just refactored the gai and key to be optional parameters instead of members of the options struct.